timemachine/cpp/src/neighborlist.cu

```diff
@@ -8,23 +8,38 @@
 
 namespace timemachine {
 
-template <typename RealType> Neighborlist<RealType>::Neighborlist(int N) : N_(N) {
+template <typename RealType>
+Neighborlist<RealType>::Neighborlist(const int N, const int K) : N_(N), K_(K), compute_full_matrix_(K > 0) {
```

Comment:
Should we change compute_full_matrix_ into a function? i.e. if K_ == 0 compute_full_matrix returns True (as opposed to storing it as a separate variable that may get out of sync during a later refactor)